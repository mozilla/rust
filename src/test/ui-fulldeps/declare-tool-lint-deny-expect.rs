// compile-flags: -Z unstable-options

#![feature(rustc_private)]

extern crate rustc_session;

use rustc_session::declare_tool_lint;

declare_tool_lint! {
    pub clippy::TEST_LINT_ALLOW,
    Allow,
    "test"
}

// This should cause an error
declare_tool_lint! {
    pub clippy::TEST_LINT_EXPECT,
    Expect,
    "test"
}
//~^^^^^ ERROR: `Expect` is not allowed as an initial level for lints

declare_tool_lint! {
    pub clippy::TEST_LINT_WARN,
    Warn,
    "test"
}

declare_tool_lint! {
    pub clippy::TEST_LINT_DENY,
    Deny,
    "test"
}

fn main() {}
