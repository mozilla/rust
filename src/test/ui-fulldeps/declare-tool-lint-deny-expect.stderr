error: `Expect` is not allowed as an initial level for lints
  --> $DIR/declare-tool-lint-deny-expect.rs:16:1
   |
LL | / declare_tool_lint! {
LL | |     pub clippy::TEST_LINT_EXPECT,
LL | |     Expect,
LL | |     "test"
LL | | }
   | |_^
   |
   = note: this error originates in the macro `$crate::validate_lint_level` (in Nightly builds, run with -Z macro-backtrace for more info)

error: aborting due to previous error

